pyi.py

```diff
@@ -925,10 +925,12 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> None:
         if _is_TypeAlias(node_annotation):
             with self.visiting_TypeAlias.enabled():
                 self.generic_visit(node)
-            if isinstance(node_target, ast.Name) and self._Y043_REGEX.match(
-                target_name
-            ):
-                self.error(node, Y043)
+            if isinstance(node_target, ast.Name):
+                if target_name.startswith("_"):
+                    if self._Y043_REGEX.match(target_name):
+                        self.error(node, Y043)
+                else:
+                    self.error(node, Y042)
```

Comment:
Minor proofing against changes to the regex. Also in theory it would be possible to trigger Y042 and Y043 on the same line (if we didn't check for the underscore in `...T` names).

```suggestion
                if not target_name.startswith("_"):
                    self.error(node, Y042)
                if self._Y043_REGEX.match(target_name):
                    self.error(node, Y043)
```